backend/webhook/consumer/pkg/worker/worker.go

```diff
@@ -0,0 +1,159 @@
+package worker
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"sync"
+
+	"github.com/beanstalkd/go-beanstalk"
+	"github.com/jpillora/backoff"
+
+	"bytes"
+	"fmt"
+
+	"log"
+	"net/http"
+	"time"
+)
+
+type Worker struct {
+	endpoint     string
+	customHeader map[string]string
+	beanstalk    *beanstalk.Conn
+	backoff      *backoff.Backoff
+	errors       []consumerError
+}
+type consumerError struct {
+	Err error
+	T   time.Time
+}
+
+type webhookConfig struct {
+	Id       string
+	Endpoint string
+	Headers  map[string]map[string]string
+	Status   string
+}
+
+func Start(
+	ctx context.Context,
+	wg *sync.WaitGroup,
+	webhookConfigStream chan string,
+	hostname, port string,
+	maxBackoff time.Duration,
+) (*Worker, error) {
+
+	conn, err := beanstalk.Dial("tcp", fmt.Sprintf("%s:%s", hostname, port))
+
+	if err != nil {
+		return nil, err
+	}
+
+	w := Worker{
+		endpoint:     "",
+		customHeader: make(map[string]string),
+		beanstalk:    conn,
+		backoff:      &backoff.Backoff{Max: maxBackoff},
+		errors:       make([]consumerError, 0),
+	}
+
+	wg.Add(2)
+	go w.Run(ctx, wg)
+	go w.updateWebhookConfig(ctx, wg, webhookConfigStream)
+	return &w, nil
+}
+
+func (w *Worker) Run(ctx context.Context, wg *sync.WaitGroup) {
+	defer wg.Done()
+	for {
+		select {
+		case <-ctx.Done():
+			log.Println("terminating worker: context canceled ")
+			return
+		default:
+			id, event, err := w.beanstalk.Reserve(1 * time.Minute)
+			if errors.Is(err, beanstalk.ErrTimeout) {
+				continue
+			}
+			if err != nil {
+				log.Println(err)
+				continue
+			}
```

Comment:
the two if could be squashed together (if not nil { if !timeout {print} continue})